### ISSUE TITLE
Add scope option to scope down relation before batch importing

### ISSUE THREAD

**Issue by neilma** [ankane/searchkick#1503]
**First**
Search existing issues to see if it’s been discussed.

**Is your feature request related to a problem? Please describe.**
It seems that the bulk_indexer gets all data in relation in batches and go through each record in the batch and disgard the record if it's not indexable, which may result in waste of resource. e.g. a whole batch may contain all non-indexable data.

**Describe the solution you'd like**
If we could provide a scope on the relation so the data/batches to be imported have the scope applied before the batch processing, it may reduce the risk of wasting time querying and computing non-indexable records.

In other words, the should_index scope can be applied to the relation level in addition to the current record level.

**Additional context**
Add any other context.

**Comment by ankane** [comment]
Hey @neilma, I believe you can use the `search_import` scope for this.

**Comment by davidwparker** [comment]
@ankane - is there a way to use `search_import` but for partial reindexes?

I want to do `.includes` for some partials but not all, and it currently can only be done all or none.
